### DISCUSSION TITLE
consider exposing "current state" of ponder indexing in graphql

### DISCUSSION THREAD

**Discussion by frolic** [ponder-sh/ponder#203] [Idea / Feature Request]
It'd be nice if there was a `__ponder` entity that exposed things like "last block processed" etc. It might also be nice to include this info on entities by default, similar to an "updated at" field?

**Comment by leez7** [comment]
contract address would also be helpful to be included

**Reply by typedarray** [reply]
Could you share more about this idea? Perhaps the idea would be to include some details from `ponder.config.ts` in a metadata field in the GraphQL schema?

Related: I have gotten requests to include some `ponder.config.ts` data in the event handler context. Would that also be helpful?

**Reply by leez7** [reply]
i have around 30 instances of the same contract and i'd like to be able to get the address so i can do on-chain calls without having to do another address lookup based on ponder.config.ts contract identifier.

re: ponder.config.ts data, yes – that would be helpful (address would help in my case)

**Reply by typedarray** [reply]
What do you mean by 30 instances of the same contract? Same ABI, but deployed to different addresses on the same network? 

Also - is your app open-source? Might be fastest for me to take a look.

**Reply by leez7** [reply]
re 30 instances – yes, the same contract deployed multiple times for multiple nft collections. it's a lending market, each contract being a protocol lending vault (minting stablecoins when users borrow against their nfts).

contracts are open source, ponder app is not – i was just playing around with it hoping to find a better alternative to the TheGraph

**Reply by typedarray** [reply]
Got it. So, if I understand your problem correctly, it seems like you should be able to use the [contract instance](https://ponder.sh/api-reference/event-handlers#readonlycontract) objects out of the box. Is there a reason they won't work?

Also, more than happy to support in any way I can. Feel free to pop into the [Ponder telegram channel](https://t.me/ponder_sh) too, might get back to you more quickly there.

**Reply by leez7** [reply]
i guess plan was to also store the address as identifier. 

re tg channel – thx, will hop in
